pkg/api/blocks/v1.go

```diff
@@ -84,9 +86,18 @@ func (bapi *BlocksAPI) Register(r *route.Router, tracer opentracing.Tracer, logg
 
 	r.Get("/blocks", instr("blocks", bapi.blocks))
 	r.Post("/blocks/mark", instr("blocks_mark", bapi.markBlock))
+	r.Get("/flags", instr("flags_disableAdminOperations", bapi.disableAdminOperationsHandler))
```

Comment:
I don't think you need to add a new API for this. We already register `api/v1/status/flags` as a common endpoint for all components in `pkg/api`

Reply:
> It would be cleaner and more consistent with other such options like disableCors, and we wouldn't need to check a string here at all
> But actually, I think we don't even need any changes in BlocksAPI here. The flag value will just directly be used in UI anyway, so we don't need a separate API. See my other comment [#6601 (comment)](https://github.com/thanos-io/thanos/pull/6601#discussion_r1289653181)

@saswatamcode @GiedriusS @yeya24  one problem that I am facing when using `/status/flags` endpoint is that I am not able to figure out how to get the flags value inside the markBlock function of v1.go. I need it so as to restrict access to the API when the flag is enabled.

Reply:
Ack! Reading the issue seems like we would need the option passed here as well. To restrict access, both via UI and API. Ignore my comment above then.

Reply:
thanks for the clarification!